org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractGenerator.java

```diff
@@ -88,15 +88,16 @@ public void setModel(T model) {
 	 * @throw exception if model invalid
 	 */
 	protected void validateModel() throws ValidationException {
-		T model = getModel();
-
 		if (model instanceof IBoundingBoxModel) {
 			IBoundingBoxModel bmodel = (IBoundingBoxModel)model;
 			// As implemented, model width and/or height can be negative,
 			// and this flips the slow and/or fast point order.
-			if (bmodel.getBoundingBox() == null) throw new ModelValidationException("The model must have a Bounding Box!", model, "boundingBox");
-	        if (bmodel.getBoundingBox().getFastAxisLength()==0)  throw new ModelValidationException("The length must not be 0!", bmodel, "boundingBox");
-	        if (bmodel.getBoundingBox().getSlowAxisLength()==0)  throw new ModelValidationException("The length must not be 0!", bmodel, "boundingBox");
+			if (bmodel.getBoundingBox() == null)
+				throw new ModelValidationException("The model must have a Bounding Box!", model, "boundingBox");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "boundingBox" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)